src/components/Router/index.js

```diff
@@ -239,7 +239,7 @@ class HistoryRouter extends Component {
       flow: newFlow || currentFlow,
     }
     if (excludeStepFromHistory) {
-      this.setState(historyState)
+      this.onHistoryChange({ state: historyState })
```

Comment:
I wonder if we should have some indirection here, to prevent confusion in the future. Basically the the instructions inside of `onHistoryChange` could be inside their own method (maybe called `updateHistoryState()`) and then both here and inside of `onHistoryChange` that new method is called.
My concern is that `onHistoryChange` is reacting to the history stack changing, so in the future we might have instructions inside that method that will work if the history stack had in fact been changed.
What do you think?

Reply:
I agree that imperative call to onHistoryChange might be a source of confusion. 

However, instead of refactoring that behaviour into proxy method to avoid that confusion, I'm proposing directly setting the state, and call the `onStepChange` callback prop if `state.step` effectively changes. 

This way `onHistoryChange` will only get called if history changes.